app/workers/preparsing.rb

```diff
@@ -27,14 +64,13 @@ def perform(genotype_id)
         system("mv #{Rails.root}/tmp/#{genotype.fs_filename}.csv #{Rails.root}/public/data/#{genotype.fs_filename}")
         logger.info "copied file"
       end
-
-    rescue
-      logger.info "nothing to unzip, seems to be a text-file in the first place"
+    else
+      system("cp #{genotype.genotype.path} #{Rails.root}/public/data/#{genotype.fs_filename}")
     end
 
     # now that they are unzipped, check if they're actually proper files
     file_is_ok = false
-    fh = File.open(genotype.genotype.path)
```

Comment:
How did this ever work?? It parses the original uploaded file the way this looks like, that can be anything before extraction